core/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregator.java

```diff
@@ -111,6 +113,12 @@ public InternalAggregation reduce(InternalAggregation aggregation, ReduceContext
         long lastValidKey = 0;
         int lastValidPosition = 0;
         int counter = 0;
+        double[] modelCoef = model.getCoef();
```

Comment:
It would be nice if the `modelCoef` could be an object so the coefficients can be named rather than being anonymous in a double[]. Maybe the object could have a method the minimiser can use which returns double[] but the model gets returns its own object from the `model.getCoef()` method?